ln-gateway/src/actor.rs

```diff
@@ -0,0 +1,326 @@
+
+        // Establish a channel to communicate with the actor
+        let (tx, mut rx): (mpsc::Sender<GatewayRequest>, mpsc::Receiver<GatewayRequest>) =
+            mpsc::channel(100);
+        let messenger = GatewayRpcSender::new(tx);
+
+        // Spawn a gateway actor thread that uses actor core when handling requests
+        let shutdown = is_actor_shutting_down.clone();
+        let handle = tokio::spawn(async move {
+            while !shutdown.clone().fetch_and(false, Ordering::Relaxed) {
+                for fetch_result in client.fetch_all_coins().await {
+                    if let Err(e) = fetch_result {
+                        debug!(error = %e, "Fetching coins failed")
+                    };
+                }
+
+                // Handle messages
+                while let Ok(msg) = rx.try_recv() {
+                    tracing::trace!("Gateway received message {:?}", msg);
+                    match msg {
+                        GatewayRequest::HtlcAccepted(inner) => {
+                            inner
+                                .handle(|inner| core.handle_htlc_incoming_msg(inner.htlc_accepted))
+                                .await;
+                        }
+                        GatewayRequest::PayInvoice(inner) => {
+                            inner
+                                .handle(|inner| core.handle_pay_invoice_msg(inner.contract_id))
+                                .await;
+                        }
+                        GatewayRequest::Balance(inner) => {
+                            inner.handle(|_| core.handle_balance_msg()).await;
+                        }
+                        GatewayRequest::DepositAddress(inner) => {
+                            inner.handle(|_| core.handle_address_msg()).await;
+                        }
+                        GatewayRequest::Deposit(inner) => {
+                            inner
+                                .handle(|deposit| core.handle_deposit_msg(deposit))
+                                .await;
+                        }
+                        GatewayRequest::Withdraw(inner) => {
+                            inner
+                                .handle(|withdraw| core.handle_withdraw_msg(withdraw))
+                                .await;
+                        }
+                    }
+                }
+            }
+        });
+
+        Ok(Self {
+            federation_id,
+            join: Some(handle),
+            is_actor_shutting_down,
+            messenger,
+        })
+    }
+
+    pub fn get_ctrl(&mut self) -> GatewayActorCtrl {
+        self.join
+            .take()
+            .expect("Someone already took control of this GatewayActor");
+
+        GatewayActorCtrl {
+            is_actor_shutting_down: Arc::clone(&self.is_actor_shutting_down),
+        }
+    }
+}
+
+pub struct GatewayActorCtrl {
+    is_actor_shutting_down: Arc<AtomicBool>,
+}
```

Comment:
What is the difference between this and the `is_actor_shutting_down` inside `GatewayActor`?